### ISSUE TITLE
role: alert when info-panel is used with dynamics

### ISSUE THREAD

**Issue by Febakke** [Altinn/app-frontend-react#1023]
### Description of the bug
We have some situations today where apps use our panel component in combination with dynamics to create an alert component that triggers based on user input. This gives us an accessebility issue where this panel should have an aria alert role so that screen readers also can get this information.

## conundrum
We cant just add an alert role on our panels because: 

> The most important thing to know about the alert role is that it's for content that is dynamically displayed, not for content that appears on page load.
[Source ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role)

One of the main uses for the panel component is to highlight information on a page, and is often used so that is visible on page load. We need to seperate dynamic alerts and those that is only used to highlight static information.

Also, this is pretty similar to our soft-validation functionality. If we expand the functionality of our panels to handle this, we have to seperate components doing almost the same thing. This might be hard to comunicate the difference between the 2. 

One solution is to in some way expand the soft validation so that it meets our appdevelopers needs. Then we can contain this to one consept that is easier to communicate. The other is to add some attribute to the panel that can be set by the app-developer when they use it with dynamics, but then we add alert functionality to a component that maybe should be used as a way of presenting static information, and by having another attribute the app-developer has to worry about, we increase the chance of accessebility issues in apps that use this feature. 

Or maybe a developer have a better suggestion on how to solve this? 😇

Small example: 
![image](https://user-images.githubusercontent.com/33222679/226847850-41460207-0e00-4e74-a730-8816e0c60f57.png)

[aria alert documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role)

## Specification
The documentation specifies that you need to be careful with the use of alert as it can be very intrusive for screen reader users. Is there a risk that it might be overused, should the developer be able to turn it of if the information is not that important. 
- [ ] Look at relevant cases
- [ ] Define guidelines
- [ ] Added to all variants (info, success and warning)
- [ ] Consider adding the alert that is specified in the designsystem [Issue](https://github.com/digdir/designsystem/issues/78)

### Steps To Reproduce

Go to an app with soft validation and test if it is read by the screen reader when it is triggered.

**Comment by StianVestli** [comment]
org/ssb

**Comment by StianVestli** [comment]
Regner med at hvis man legger på aria-live på denne så vil dette være på alle varianter av denne boksen. dvs alle disse:
info
success
warning

**Comment by Febakke** [comment]
Ja, det tenker jeg. Oppdaterer issuet.

Det viktige her er at den kun får den når den følger [alert mønsteret. ](https://www.w3.org/WAI/ARIA/apg/patterns/alert/) Hvis det er en statisk melding, så burde den ikke ha den rollen, kun om den trigges av noe som brukeren må få med seg.

**Comment by StianVestli** [comment]
Med alert mønsert er det hvis vi legge den med vis skjul funksjonalitet ? At den bare at vil ha aria-live?

**Comment by StianVestli** [comment]
Blir denne saken lagt til i noen sprinter i nær framtid?

**Comment by olemartinorg** [comment]
Diskusjon om denne, og hvorfor den er satt til blocked:
- https://altinndevops.slack.com/archives/C045EB3JA9X/p1681392993791739

**Comment by tjololo** [comment]
Closing as we will resolve this in #726
